httpie_aws_authv4.py

```diff
@@ -87,36 +100,81 @@ class AWSv4AuthPlugin(AuthPlugin):
     prompt_password = False
 
     def get_auth(self, username=None, password=None):
-        # To remain consistant with AWS tools, boto3 credential store is used
-        #   to retrieve the users API keys.  This means environment variables
-        #   take precedent over ~/.aws/credentials and IAM roles.  credentials
-        #   can be provided on the CLI using the -a flag
-        #       eg, -a <access_key>:<secret_access_key>
-
-        # An attempt is made to try and determine the region, endpoint and
-        #   service from a given request URL.  If you are using a custom domain
-        #   this is no doubt going fail and the user will need to input the neccessary
-        #   information using the `-a` flag.
-
-        # The format for the `-a` flag is:
-        #   "service:region:endpoint"
+        """
+        To remain consistant with AWS tools, boto3 credential store is used
+        to retrieve the users API keys.  This means environment variables
+        take precedent over ~/.aws/credentials and IAM roles.  credentials
+        can be provided on the CLI using the -a flag eg,
+            -a <access_key>:<secret_access_key>
+
+        You can also provide aditional parameters using following syntax:
+            -a access_key=<access_key>:secret_key=<secret_key>:
+            profile=<profile>:domain=<domain>:service=<service>:region=<region>
+
+        or short equvivalent:
+            -a ak=<access_key>:sk=<secret_key>:p=<profile>:d=<domain>:
+            s=<service>:r=<region>
+
+        For example:
+            -a s=executeapi:r=eu-west-1
+
+        An attempt is made to try and determine the region, endpoint and
+        service from a given request URL if you are not supplying this data as above.
+        If you are using a custom domain this will fail and service plus region need
+        to be supplied in the -a flag.
+        """
 
         access_key = None
         secret_key = None
         domain = None
         profile = None
+        service = None
+        region = None
 
         if self.raw_auth is not None:
-            parts = self.raw_auth.split(':')
-            if len(parts) >= 2:
-                if parts[0] == 'profile':
-                    profile = parts[1]
-                else:
+            try:
+                params = dict(x.split('=') for x in self.raw_auth.split(':'))
+                if 'access_key' in params:
+                    access_key = params['access_key']
```

Comment:
This section can be greatly simplified by using dict unpacking commented below.  Alias/short-name arguments could be handled with a simple map that is iterated over, maybe something like:

```
alias_params = {"ak": "access_key", "sk": "secret_key"} # map of alias -> full param name

for alias, full in alias_params.items():
    if alias in params:
        params[full] = params[alias]
        del params[alias]

```